Add credential dataset version (#472)
openid/OpenID4VCI: openid-4-verifiable-credential-issuance-1_0.md

Credential Dataset:
:  A set of one or more claims about a subject, provided by a Credential Issuer.

Credential Dataset Identifier

> : A persistent identifier that refers to a specific version of a Credential Dataset. It remains stable across multiple instances of a Credential that share the same set of claim values, even if they differ in cryptographic proofs. When the claim values in the dataset change, a new Credential Dataset Identifier is assigned. This enables Wallets to detect meaningful changes to the underlying data and to distinguish between different versions of a Credential Dataset issued under the same Credential Configuration.

Reviewer: ```suggestion
: A unique identifier that refers to a specific version of a Credential Dataset. This identifier remains stable across multiple instances of a Credential that share the same set of claim values, even if they differ in cryptographic proofs. When the claim values in the dataset change, a new Credential Dataset Identifier is assigned. This identifier enables Wallets to detect changes to the underlying data and to distinguish between Credentials issued with different versions of a Credential Dataset under the same Credential Configuration.
```

   * `credential`: REQUIRED. Contains one issued Credential. It MAY be a string or an object, depending on the Credential Format. See Appendix A for the Credential Format-specific encoding requirements.
* `transaction_id`: OPTIONAL. String identifying a Deferred Issuance transaction. This parameter is contained in the response if the Credential Issuer cannot immediately issue the Credential. The value is subsequently used to obtain the respective Credential with the Deferred Credential Endpoint (see (#deferred-credential-issuance)). It MUST not be used if the `credentials` parameter is present. It MUST be invalidated after the Credential for which it was meant has been obtained by the Wallet.
* `notification_id`: OPTIONAL. String identifying one or more Credentials issued in one Credential Response. It MUST be included in the Notification Request as defined in (#notification). It MUST not be used if the `credentials` parameter is not present.

> * `credential_dataset_id`: OPTIONAL.  An opaque string containing the Credential Dataset Identifier associated with the returned Credential(s). This allows Wallets to detect changes to the underlying Credential Dataset across different Credential Responses.

Reviewer: Personally I think this feature is important enough that this identifier should be mandatory to return otherwise it makes it very difficult for credential update/refresh to be supported. If its left optional we need to discuss how it gets signalled whether this feature is supported and what a wallet is suppose to assume when it comes to updating/refreshing credentials.
